litestar/contrib/sqlalchemy/repository.py

```diff
@@ -351,9 +351,7 @@ async def list_and_count(
         statement = kwargs.pop("base_select", self.statement)
         statement = statement.add_columns(
             over(
-                sql_func.count(
-                    self.model_type.id,
-                ),
+                sql_func.count(getattr(self.model_type, self.id_attribute)),
```

Comment:
```suggestion
                sql_func.count(self.get_id_attribute_value(self.model_type)),
```

Reply:
Is the reason you aren't using this method due to the type signature of `get_id_attribute_value`? We can change that to `T | type[T]`, or even `Any` as it will fail if `item` has not got the attribute.

Reply:
The real reason was a lack of understanding. But yes, this does raise a type error. I think `T | type[T]` will suffice, let me check